var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -255,18 +264,28 @@ def patch(self):
                         '-I{0}'.format(spec['zlib'].prefix.include),
                         'gcc/Makefile.in')
 
+    # https://gcc.gnu.org/install/configure.html
     def configure_args(self):
         spec = self.spec
 
         # Generic options to compile GCC
         options = [
+            # Distributor options
+            '--with-pkgversion=Spack GCC',
+            '--with-bugurl=https://github.com/spack/spack/issues',
+            # Xcode 10 dropped 32-bit support
             '--disable-multilib',
+            # Don't bother with full 3-stage bootstrapping process
+            # Builds much faster, but doesn't test build as well
+            # See https://gcc.gnu.org/install/build.html
+            '--disable-bootstrap',
```

Comment:
Are we sure we want to disable 3-stage bootstrapping? Build time might be long, but compiler is self-consistent and uses the latest optimizations available in the back-end. Also: do we get additional dependencies on the compiler being used for stage 1 in this case compared to the 3 stage bootstrapping?

Reply:
> Also: do we get additional dependencies on the compiler being used for stage 1 in this case compared to the 3 stage bootstrapping?

As far as I know, yes.  I don't think it is a problem if it's the system (native) compiler, but may be if it is not.

> Are we sure we want to disable 3-stage bootstrapping?

 I strongly disagree with disabling bootstrap by default, was waiting to respond to see what other reviewers said or did not...I propose we enable a variant so that Adam and others who need it can disable it with relative convenience. 

The way I read the build docs is (see attached patch, stating to confirm):

- If not cross-compiling, and if `--disable-bootstrap` not specified:
    - then `--enable-bootstrap` implied, and `make` does `bootstrap` by default.
- If cross-compiling, or `--disable-bootstrap` specified:
    - then `make` does **not** `bootstrap`
- Meaning as far as I can tell, homebrew removed `make bootstrap` because it wasn't necessary in the first place.

I think it would be nice to have a multi-value `[default, profiled, disabled]` variant for this.  Proposed "patch" (off `develop`, sorry adam, not applicable to this branch b/c it's incomplete anyway...):

[spacky.patch.txt](https://github.com/spack/spack/files/3724332/spacky.patch.txt)

Issues:

- The multi-value thing causes some problems though, I don't know how to `conflicts` against it (e.g., with `+nvptx`, not `%gcc`, see TODO in patch).
- What's the story with spack and cross-compiling?  Is it already supported, or in the roadmap?  If it's ever desired, and I understand the GCC build docs correctly, we should simply not `--enable-bootstrap` -- only disable it.

Reply:
I'm fine with removing this if it's controversial. I was running into weird problems where I would run the build overnight and 9+ hrs later it was still running. The last time it only took 2 hrs, so I'm not sure what was going on. But 30 min is better than 2 hrs, assuming bootstrapping isn't important. It would be nice if we could do a 2-stage bootstrap...